### PR TITLE
Admin frontend

### PR DESCRIPTION
**Pull request by strdavis**
- [x] Implement global layout and styling.
- [x] Rename wage to applications.
- [x] Always filter out drafts.
- [x] Implement catchment handling and state management.
- [x] Implement applications datagrid.
- [x] Implement applications status sidebar.
- [x] Implement catchment dropdown (with default styling).
- [x] Implement catchment label.
- [x] Implement Notify button, styling only.
- [x] Implement PDF button, styling only. 
- [x] Implement claims list
- [x] Synchronize catchment across lists
- [x] Implement calculator button, styling only
- [x] Implement Move button, styling only
- [x] Make list rows keyboard navigable
- [x] Make list rows keyboard selectable  
- [x] Visually indicate when rows receives keyboard focus
- [x] Visually indicate when Notify button receives keyboard focus
- [x] Visually indicate when Catchment Dropdown receives keyboard focus 
- [x] Make list rows visible to screen reader as buttons, and also as readable table rows.
- [x] Ensure all content visible to screen reader.
- [x] Improve aria labels.
- [x] Clean up and refactor applications endpoints.
- [x] Clean up and refactor claims endpoints.

### CODE REVIEW

File: packages/admin-client/src/Applications/ApplicationList.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+export const ApplicationList = (props: any) => {
+    const cc = useContext(CatchmentContext)
+    const unselectAll = useUnselectAll("applications")
+
+    useEffect(() => {
+        unselectAll()
+    }, [cc.catchment])
+
+    const handleRowClick = (id: Identifier, resource: string, record: any) => {
+        console.log("row click")
```
Comment (brankoEISI) [inline packages/admin-client/src/Applications/ApplicationList.tsx:28]: Should remove the console logs and put the commented out temporary click functionality back in

File: packages/admin-client/src/Claims/ClaimList.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    const cc = useContext(CatchmentContext)
+    const unselectAll = useUnselectAll("claims")
+    const [statusFilter, setStatusFilter] = useState(claimStatusFilters["All"])
+
+    useEffect(() => {
+        unselectAll()
+    }, [cc.catchment])
+
+    const handleRowClick = (id: Identifier, resource: string, record: any) => {
+        console.log("row click")
```
Comment (brankoEISI) [inline packages/admin-client/src/Claims/ClaimList.tsx:28]: remove console logs

File: packages/admin-client/src/common/components/CalculatorButtonField/CalculatorButtonField.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { COLOURS } from "../../../Colours"
+
+const CalculatorButtonField: React.FC = () => {
+    const record = useRecordContext()
+
+    return record.status === "Processing" ? (
+        <Button
+            onClick={(event) => {
+                event?.stopPropagation()
+                console.log("calculator button")
```
Comment (brankoEISI) [inline packages/admin-client/src/common/components/CalculatorButtonField/CalculatorButtonField.tsx:14]: remove console logs

File: packages/admin-client/src/common/components/PdfButtonField/PdfButtonField.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import { COLOURS } from "../../../Colours"
+
+const PdfButtonField: React.FC = () => {
+    const record = useRecordContext()
+
+    return (
+        <Button
+            onClick={(event) => {
+                event?.stopPropagation()
+                console.log("pdf button")
```
Comment (brankoEISI) [inline packages/admin-client/src/common/components/PdfButtonField/PdfButtonField.tsx:14]: console logs

File: packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    const access = localStorage.getItem("true")
+    if (provider === "BCEID" && permissions !== null) {
+        JSON.parse(permissions).forEach((item: { catchmentNo: number; location: string }) => {
+            result?.push({
+                id: item.catchmentNo,
+                name: createCatchmentName(item.catchmentNo, item.location)
+            })
+        })
+    } else if (provider === "IDIR" && access === "true") {
+        for (let i = 1; i <= 45; i += 1) {
```
Comment (brankoEISI) [inline packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx:22]: put the 45 into a constant at the top of the file, something like 
```javascript
const NUM_CATCHMENTS = 45
```

File: packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
+import { createContext, useState, useEffect, useMemo } from "react"
+
+const createCatchmentName = (catchmentNo: number, location: string) => {
+    let catchmentName = catchmentNo < 10 ? "0" : ""
+    return catchmentName + catchmentNo.toString() + " - " + location
+}
+
+const getCatchments = () => {
+    // eslint-disable-next-line prefer-const
+    let result: any[] | undefined = []
```
Comment (brankoEISI) [inline packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx:10]: result will never be undefined if you're instantiating it as an empty array and only doing pushes on it

File: packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    const provider = localStorage.getItem("provider")
+    const permissions = localStorage.getItem("permissions")
+    const access = localStorage.getItem("true")
+    if (provider === "BCEID" && permissions !== null) {
+        JSON.parse(permissions).forEach((item: { catchmentNo: number; location: string }) => {
+            result?.push({
```
Comment (brankoEISI) [inline packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx:16]: don't need the ? null check, same reason as above

File: packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+                id: item.catchmentNo,
+                name: createCatchmentName(item.catchmentNo, item.location)
+            })
+        })
+    } else if (provider === "IDIR" && access === "true") {
+        for (let i = 1; i <= 45; i += 1) {
+            result?.push({ id: i, name: `${i}` })
```
Comment (brankoEISI) [inline packages/admin-client/src/common/contexts/CatchmentContext/CatchmentContext.tsx:23]: don't need null check

**Review by brankoEISI — CHANGES_REQUESTED** [review]

File: packages/admin-api/src/services/claims.service.ts [hunk 9]
```diff
@@ -33,93 +24,34 @@ export const getAllClaims = async (perPage: number, currPage: number, filters: a
...
-    }
-
-    // 2. Check if the user has permission to edit this claim
-    if (claims[0].catchmentno in permission || permission[0] === "*") {
-        // 3. Update the claim
-        const result = await knex("claims")
+export const updateClaim = async (id: number, status: string | null, form: any) => {
+    let numUpdated = 0
+    if (form && form.userInfo) {
+        numUpdated = await knex("claims")
```
[the remark this answers is not in the document]
Comment (strdavis) [reply packages/admin-api/src/services/claims.service.ts:50]: It does appear to return the number of records updated.

**Review by brankoEISI — APPROVED** [review]

**Pull request merged**